parlai/core/torch_ranker_agent.py

```diff
@@ -176,6 +177,27 @@ def __init__(self, opt, shared=None):
                 self.model, device_ids=[self.opt['gpu']], broadcast_buffers=False
             )
 
+    def interactive_mode(self, mode, shared=False):
+        if mode:
+            if not shared:
+                # Only print in the non-shared version.
+                print("[" + self.id + ': full interactive mode on.' + ']')
+            self.eval_candidates = 'fixed'
+            self.ignore_bad_candidates = True
+            self.encode_candidate_vecs = True
+            if self.opt['fixed_candidates_path'] is None:
+                # Attempt to get a standard candidate set for the given task
+                # TODO(fill this bit in and built a cand set if it doesn't exist)
+                # path = self.task_candidate_set_path(self.opt['task'])
+                if not shared:
+                    print("[setting fixed_candidates path to: ] " + path)
+                self.fixed_candidates_path = path
+        else:
+            self.eval_candidates = self.opt['eval_candidates']
```

Comment:
why is `self.eval_candidates` set here and `self.candidates` set in the init? this is a bit confusing

Reply:
that one isn't affected by interactive mode, but i can move it if you want

Reply:
yeah i think we should move it to the same place that candidates are set so people can see more clearly how this works. then in the case of interactive_mode we just override that

Reply:
ok